### DISCUSSION TITLE
Sensor Update Period

### DISCUSSION THREAD

**Discussion by JamesWJPS** [macxq/foxess-ha#62] [General]
Hi. When I look at the sensors they only seem to pull about once an hour. Is there a way of changing the frequency as I want to use the data to drive a switch for solar changing my car. 

I have looked and seen the scan interval, but I don’t know if I am doing to wrong. Reducing it seems to make no difference. 

Otherwise a great solution. 

Regards

James

**Comment by vicdavery** [comment]
I came here just to mention this. It appears from the code that the interval is supposed to be 5 minutes, but I frequently don't see an update for an hour or more. 
No connectivity issues are reported in the logs. 

Any advice would be greatly appreciated. I know an hour isn't a long while, but I'd like to know why I don't get the updates.
![image](https://user-images.githubusercontent.com/2733869/187037854-692864c0-460c-46b7-8075-d6014162b6aa.png)

**Comment by Daibutt** [comment]
SoC doesn't change frequently, so if HA doesn't see a change it won't update the number, because there's no need to.
Is this the same for other sensors tho too? i.e. PV, load bat charge etc...?

**Reply by vicdavery** [reply]
I provided a bad example then. But, yes it is the same for more real-time sensors, e.g.
![image](https://user-images.githubusercontent.com/2733869/187068392-dd804fcf-d7c0-409d-8737-17463c21098a.png)

**Reply by Daibutt** [reply]
Ah OK, I'm not really sure then sorry. 
Not currently using this integration as I have the local modbus connection, but when I was using it the sensors used to update at the scan interval set in the code.
Do you know what the scan interval is set at on your inverter? This will have to be the same (or lower) than the scan interval in the HA code as that's what sends the data to Foxcloud (which this integration then accesses).
![image](https://user-images.githubusercontent.com/69276073/187072197-cb647c6b-27fb-4fc7-bffa-1ed4df757360.png)

I think this setting is also available in the app, not sure about directly from inverter screen.

**Comment by vicdavery** [comment]
Thanks for getting back to me.
I'm not sure where that screen is from, I can't see anything similar in the
app or the fox website.

I can see that the graphs on the fox website have updates every 5 minutes.
So I think the data is getting uploaded correctly and it must be something
on the HA side which is failing to get the info.

**Reply by Daibutt** [reply]
OK, yes if the website is updating at 5 mins then it does sound like an HA issue as you say. Maybe @macxq can help understand why.
